ci-operator/step-registry/hypershift/hive/create/hostedcluster/hypershift-hive-create-hostedcluster-commands.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+AWS_GUEST_INFRA_CREDENTIALS_FILE="/etc/hypershift-ci-jobs-awscreds/credentials"
+DEFAULT_BASE_DOMAIN=ci.hypershift.devcluster.openshift.com
+
+echo Generating pull secret to current build farm
+oc registry login --to=${SHARED_DIR}/pull-secret-build-farm.json
+echo "Set KUBECONFIG to Hive cluster"
+export KUBECONFIG=/var/run/hypershift-workload-credentials/kubeconfig
+
+if [[ $HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT == "true" ]]; then
+  AWS_GUEST_INFRA_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+  DEFAULT_BASE_DOMAIN=origin-ci-int-aws.dev.rhcloud.com
+fi
+DOMAIN=${HYPERSHIFT_BASE_DOMAIN:-$DEFAULT_BASE_DOMAIN}
+RELEASE_IMAGE=${HYPERSHIFT_HC_RELEASE_IMAGE:-$RELEASE_IMAGE_LATEST}
+
+# We don't have the value of HYPERSHIFT_RELEASE_LATEST when we set CONTROLPLANE_OPERATOR_IMAGE so we
+# have to use a hack like this.
+[[ ${CONTROLPLANE_OPERATOR_IMAGE} = "LATEST" ]] && CONTROLPLANE_OPERATOR_IMAGE="${HYPERSHIFT_RELEASE_LATEST}"
+
+HASH="$(echo -n $PROW_JOB_ID|sha256sum)"
+CLUSTER_NAME=${HASH:0:20}
+INFRA_ID=${HASH:20:5}
+echo "Using cluster name $CLUSTER_NAME and infra id $INFRA_ID"
+echo "CLUSTER_NAME=$CLUSTER_NAME" > ${SHARED_DIR}/hosted_cluster.txt
+echo "INFRA_ID=$INFRA_ID" >> ${SHARED_DIR}/hosted_cluster.txt
+
+if [[ -f ${SHARED_DIR}/pull-secret-build-farm.json ]]; then
+  jq -s '.[0] * .[1]' ${SHARED_DIR}/pull-secret-build-farm.json /etc/ci-pull-credentials/.dockerconfigjson > /tmp/pull-secret.json
+else
+  cp /etc/ci-pull-credentials/.dockerconfigjson /tmp/pull-secret.json
+fi
+
+echo "$(date) Creating HyperShift cluster ${CLUSTER_NAME}"
```

Comment:
What prevents more than a certain number of hosted clusters from being created? I understand that 80 is the recommended maximum at the moment. Seems the workflow should hold a lease for a hosted cluster and the account in which they are trying to establish their guest nodes.

Reply:
The workflow requested a lease. The lease is managed by boskos and set to 80 (I confirmed with HypreShift team).

Reply:
@bbguimaraes  can you confirm if what I do is correct (request lease at create step) and can fulfill our goal (to limit the hostedcluster to 80)?